### ISSUE TITLE
Not logging to Kibana with Serilog when field name is "cronExpression"

### ISSUE THREAD

**Issue by UnderHeavyDevelopment** [serilog/serilog-sinks-elasticsearch#132]
**Does this issue relate to a new *feature* or an existing *bug*?**
- [x] Bug
- [ ] New Feature

**What version of Serilog.Sinks.Elasticsearch is affected?  Please list the related NuGet package.**
5.3.0

**What is the target framework and operating system? See [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**

- [?] netCore 2.0
- [?] netCore 1.0
- [?] 4.7
- [x] 4.6.x
- [?] 4.5.x

**Kibana version**:
5.6.3

**Elasticsearch version**:
5.6.3

**Other version**:
Serilog 2.5.0
.NET Framework 4.6.1

**Description of the problem including expected versus actual behavior**:
When logging with Serilog and field name is "cronExpression", than event is not shown in Kibana log viewer (all filters cleared). However event is logged to other sinks, like file or console.

**Steps to reproduce**:
Execute following code:

```
using Serilog;
using Serilog.Sinks.Elasticsearch;

public class Test
{
    public void Log()
    {
           var logger = new LoggerConfiguration()
                .WriteTo.Elasticsearch(new ElasticsearchSinkOptions("url")
                {
                    AutoRegisterTemplate = true,
                    CustomFormatter = new ExceptionAsObjectJsonFormatter(renderMessage: true)
                }).CreateLogger();

            string a = "test";
            logger.Information("! {cronExpression}", a); // this is not logged to Kibana
            logger.Information("!! {d}", a); // this is logged to Kibana
    }
}
```

**Comment by mivano** [comment]
Did you look at the Elasticseach logs themselves (so on the server, in the logs folder)? Or enabled the selflog functionality for Serilog and see if it throws an exception?

**Comment by UnderHeavyDevelopment** [comment]
The only errors in Elasticsearch are about creating index on "cronExpression" field:

TL;DR
```
org.elasticsearch.index.mapper.MapperParsingException: failed to parse [fields.cronExpression]
Caused by: java.lang.IllegalArgumentException: Invalid format: "0 0 0/1 ? * * *" is malformed at " 0 0/1 ? * * *"
```

Full
```
[2017-11-28T16:14:33,077][DEBUG][o.e.a.b.TransportShardBulkAction] [n3WPM71] [logstash-2017.11.28][0] failed to execute bulk item (index) BulkShardRequest [[logstash-2017.11.28][0]] containing [3] requests
org.elasticsearch.index.mapper.MapperParsingException: failed to parse [fields.cronExpression]
        at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:298) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:468) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseValue(DocumentParser.java:591) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:396) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:465) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseObject(DocumentParser.java:484) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:383) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.internalParseDocument(DocumentParser.java:93) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:66) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:277) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.shard.IndexShard.prepareIndex(IndexShard.java:530) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.shard.IndexShard.prepareIndexOnPrimary(IndexShard.java:507) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.bulk.TransportShardBulkAction.prepareIndexOperationOnPrimary(TransportShardBulkAction.java:458) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.bulk.TransportShardBulkAction.executeIndexRequestOnPrimary(TransportShardBulkAction.java:466) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.bulk.TransportShardBulkAction.executeBulkItemRequest(TransportShardBulkAction.java:146) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardOperationOnPrimary(TransportShardBulkAction.java:115) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.bulk.TransportShardBulkAction.shardOperationOnPrimary(TransportShardBulkAction.java:70) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryShardReference.perform(TransportReplicationAction.java:975) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryShardReference.perform(TransportReplicationAction.java:944) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.ReplicationOperation.execute(ReplicationOperation.java:113) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onResponse(TransportReplicationAction.java:345) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.onResponse(TransportReplicationAction.java:270) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$1.onResponse(TransportReplicationAction.java:924) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$1.onResponse(TransportReplicationAction.java:921) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.shard.IndexShardOperationsLock.acquire(IndexShardOperationsLock.java:151) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.shard.IndexShard.acquirePrimaryOperationLock(IndexShard.java:1659) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction.acquirePrimaryShardReference(TransportReplicationAction.java:933) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction.access$500(TransportReplicationAction.java:92) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$AsyncPrimaryAction.doRun(TransportReplicationAction.java:291) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryOperationTransportHandler.messageReceived(TransportReplicationAction.java:266) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.action.support.replication.TransportReplicationAction$PrimaryOperationTransportHandler.messageReceived(TransportReplicationAction.java:248) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:69) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.transport.TransportService$7.doRun(TransportService.java:644) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:638) [elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) [elasticsearch-5.6.3.jar:5.6.3]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:?]
        at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: java.lang.IllegalArgumentException: Invalid format: "0 0 0/1 ? * * *" is malformed at " 0 0/1 ? * * *"
        at org.joda.time.format.DateTimeParserBucket.doParseMillis(DateTimeParserBucket.java:187) ~[joda-time-2.9.5.jar:2.9.5]
        at org.joda.time.format.DateTimeFormatter.parseMillis(DateTimeFormatter.java:826) ~[joda-time-2.9.5.jar:2.9.5]
        at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parse(DateFieldMapper.java:240) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.DateFieldMapper.parseCreateField(DateFieldMapper.java:465) ~[elasticsearch-5.6.3.jar:5.6.3]
        at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:287) ~[elasticsearch-5.6.3.jar:5.6.3]
        ... 40 more
```

Today I run same code, and events are logged and displayed in Kibana. No errors in Elasticsearch log about "cronExpression" index error.

**Comment by mivano** [comment]
Basically, in the index of yesterday, you had a mapping which specified a different format for that field name. Might be because you indexed another type in it (like an int or an object). So ES assumes it will be int (or object) and makes a mapping representing this. Next time you index it with a different type, it will fail (silently unfortunately for now). 

Since it will create a new index per day, it will have its default mapping again. And your field name is then mapped differently. 

So see what the mapping was for the 'faulty' index and compare with the one from today.

**Comment by UnderHeavyDevelopment** [comment]
Thank you for explanation. Exception was thrown from 
`org.joda.time.format.DateTimeParserBucket.doParseMillis(DateTimeParserBucket.java:187)`
then from
`org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parse(DateFieldMapper.java:240)`
so I assume field was of DateTime type, but string was passed (first space character caused exception).
Might be as you stated, that field was reindexed for different type next day. However all was automatic, I didn't set field type explicitly. 
It is definitely ElasticSearch issue.
